h5py/_proxy.pyx

```diff
@@ -138,7 +148,31 @@ cdef herr_t dset_rw(hid_t dset, hid_t mtype, hid_t mspace, hid_t fspace,
                 if read:
                     h5py_copy(mtype, mspace, back_buf, progbuf, H5PY_GATHER)
 
-            if read:
+            if H5Tis_variable_str(mtype):  # numpy.dtypes.StringDType
+                assert NUMPY_GE2
+                if read:
+                    H5Dread(dset, dstype, cspace, fspace, dxpl, conv_buf)
+                    if not H5Tis_variable_str(dstype):
+                        # Convert fixed-width bytes to char** in place.
+                        # When dstype is already char**, H5Tconvert is a very expensive no-op.
+                        H5Tconvert(dstype, mtype, npoints, conv_buf, back_buf, dxpl)
```

Comment:
We could achieve a 25% speedup by writing a specialized variant of vstrings_scatter instead. I chose not to as I believe this should be a rather niche use case so I preferred to avoid code duplication.